pkg/build/builder/sti.go

```diff
@@ -92,8 +92,14 @@ func (s *S2IBuilder) Build() error {
 	if s.build.Spec.Strategy.SourceStrategy == nil {
 		return fmt.Errorf("the source to image builder must be used with the source strategy")
 	}
-
+	var runtimeImageName string
 	var push bool
+	runtimeImage := s.build.Spec.Strategy.SourceStrategy.RuntimeImage
+	runtimeArtifacts := copyArtifactSourceList(s.build.Spec.Strategy.SourceStrategy.RuntimeArtifacts)
+
+	if runtimeImage != nil && len(runtimeImage.Name) > 0 {
+		runtimeImageName = runtimeImage.Name
+	}
```

Comment:
didn't push the latest changes. i think this should be a part of the buildconfig validation than it should be here.